### ISSUE TITLE
Can't run tikv inside container

### ISSUE THREAD

**Issue by jetora** [tikv/tikv#11296]
## Bug Report

Can't run tikv inside container (cgroups)

### What version of TiKV are you using?
5.2.2

### What operating system and CPU are you using?
vendor_id       : GenuineIntel
model name      : Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz

CentOS Linux release 7.6.1810 (Core)

[root@tikv1 tikv-20160]# cat /proc/version
Linux version 3.10.0-693.5.2.el7.jd_3313.x86_64 (root@localhost.localdomain) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) #1 SMP Wed May 2 14:35:41 CST 2018
[root@tikv1 tikv-20160]# uname -a
Linux tikv1 3.10.0-693.5.2.el7.jd_3313.x86_64 #1 SMP Wed May 2 14:35:41 CST 2018 x86_64 x86_64 x86_64 GNU/Linux
[root@tikv1 tikv-20160]# uname -r
3.10.0-693.5.2.el7.jd_3313.x86_64

### Steps to reproduce
run it on container, directly or using tiup playground

### What did you expect?
run successfully

### What did happened?
crash

tiup error below:
```
Error: init config failed: xxx:20160: executor.ssh.execute_failed: Failed to execute command over SSH for 'tidb@xxx:22' {ssh_stderr: thread 'main' panicked at 'assertion failed: ret.insert(key, value).is_none()', components/tikv_util/src/sys/cgroup.rs:184:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
, ssh_stdout: , ssh_command: export LANG=C; PATH=$PATH:/bin:/sbin:/usr/bin:/usr/sbin /export/tidb-deploy/tikv-20160/bin/tikv-server --config-check --config=/export/tidb-deploy/tikv-20160/conf/tikv.toml --pd "" --data-dir "/export/tidb-data/tikv-20160"}, cause: Process exited with status 101: check config failed
```

with manual startup
```
[root@tikv1 tikv-20160]# export RUST_BACKTRACE=1
[root@tikv1 tikv-20160]# export MALLOC_CONF="prof:true,prof_active:false"
[root@tikv1 tikv-20160]# bin/tikv-server \
>     --addr "0.0.0.0:20160" \
>     --advertise-addr "xxx:20160" \
>     --status-addr "0.0.0.0:20180" \
>     --advertise-status-addr "xxx:20180" \
>     --pd "xxx" \
>     --data-dir "/export/tidb-data/tikv-20160" \
>     --config conf/tikv.toml \
>     --log-file "/export/tidb-deploy/tikv-20160/log/tikv.log"
thread 'main' panicked at 'assertion failed: ret.insert(key, value).is_none()', components/tikv_util/src/sys/cgroup.rs:184:17
stack backtrace:
   0: rust_begin_unwind
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:515:5
   1: core::panicking::panic_fmt
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/panicking.rs:92:14
   2: core::panicking::panic
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/panicking.rs:50:5
   3: tikv_util::sys::cgroup::parse_mountinfos_v1
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:184:17
   4: tikv_util::sys::cgroup::cgroup_mountinfos_v1
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:174:5
   5: tikv_util::sys::cgroup::CGroupSys::new
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:66:44
   6: <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref::__static_ref_initialize
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/mod.rs:25:49
   7: core::ops::function::FnOnce::call_once
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/ops/function.rs:227:5
   8: lazy_static::lazy::Lazy<T>::get::{{closure}}
             at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:31:29
   9: std::sync::once::Once::call_once::{{closure}}
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:261:41
  10: std::sync::once::Once::call_inner
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:418:21
  11: std::sync::once::Once::call_once
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:261:9
  12: lazy_static::lazy::Lazy<T>::get
             at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:30:9
  13: <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref::__stability
             at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:142:21
  14: <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref
             at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:144:17
  15: tikv_util::sys::SysQuota::cpu_cores_quota
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/mod.rs:33:28
  16: <tikv::config::UnifiedReadPoolConfig as core::default::Default>::default
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:1705:23
  17: <tikv::config::ReadPoolConfig as core::default::Default>::default
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2008:5
  18: <tikv::config::TiKvConfig as core::default::Default>::default
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2434:23
  19: <tikv::config::_IMPL_DESERIALIZE_FOR_TiKvConfig::<impl serde::de::Deserialize for tikv::config::TiKvConfig>::deserialize::__Visitor as serde::de::Visitor>::visit_map
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2308:28
  20: <&mut toml::de::Deserializer as serde::de::Deserializer>::deserialize_any
             at /rust/registry/src/github.com-1ecc6299db9ec823/toml-0.5.7/src/de.rs:220:19
  21: <&mut toml::de::Deserializer as serde::de::Deserializer>::deserialize_struct
             at /rust/registry/src/github.com-1ecc6299db9ec823/toml-0.5.7/src/de.rs:315:9
  22: tikv::config::_IMPL_DESERIALIZE_FOR_TiKvConfig::<impl serde::de::Deserialize for tikv::config::TiKvConfig>::deserialize
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2308:28
  23: tikv::config::TiKvConfig::from_file::{{closure}}
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2829:17
  24: tikv::config::TiKvConfig::from_file
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2823:9
  25: tikv_server::main::{{closure}}
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/cmd/tikv-server/src/main.rs:162:13
  26: core::option::Option<T>::map_or_else
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/option.rs:873:24
  27: tikv_server::main
             at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/cmd/tikv-server/src/main.rs:159:22
  28: core::ops::function::FnOnce::call_once
             at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

```
[root@tikv1 tikv-20160]# export RUST_BACKTRACE=full
[root@tikv1 tikv-20160]# bin/tikv-server     --addr "0.0.0.0:20160"     --advertise-addr "xxx:20160"     --status-addr "0.0.0.0:20180"     --advertise-status-addr "xxx:20180"     --pd "xxx"     --data-dir "/export/tidb-data/tikv-20160"     --config conf/tikv.toml     --log-file "/export/tidb-deploy/tikv-20160/log/tikv.log"
thread 'main' panicked at 'assertion failed: ret.insert(key, value).is_none()', components/tikv_util/src/sys/cgroup.rs:184:17
stack backtrace:
   0:     0x562540ac0233 - std::backtrace_rs::backtrace::libunwind::trace::h99dbb39dca18857d
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x562540ac0233 - std::backtrace_rs::backtrace::trace_unsynchronized::h832861927e9cfedf
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x562540ac0233 - std::sys_common::backtrace::_print_fmt::h3d18154c77dcf310
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x562540ac0233 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he312f4ad5b9bb346
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x56253fa3227c - core::fmt::write::h9a6d9c74526a6c1b
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/fmt/mod.rs:1115:17
   5:     0x562540abe574 - std::io::Write::write_fmt::h6aced00850e8186f
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/io/mod.rs:1665:15
   6:     0x562540abf2eb - std::sys_common::backtrace::_print::h65d996766de40da4
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x562540abf2eb - std::sys_common::backtrace::print::h40df9727e635f303
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x562540abf2eb - std::panicking::default_hook::{{closure}}::hd2da4327dea91a51
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:208:50
   9:     0x562540abe205 - std::panicking::default_hook::h3d55120ad6ada158
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:225:9
  10:     0x562540abe205 - std::panicking::rust_panic_with_hook::hf85dd0bb545e3b55
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:622:17
  11:     0x562540ad813e - std::panicking::begin_panic_handler::{{closure}}::h736ae969434da9fa
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:517:13
  12:     0x562540ad80dc - std::sys_common::backtrace::__rust_end_short_backtrace::h6133bb80b1d6c3e0
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x562540ad808d - rust_begin_unwind
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/panicking.rs:515:5
  14:     0x56253f7dec70 - core::panicking::panic_fmt::hcf5f6d96e1dd7099
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/panicking.rs:92:14
  15:     0x56253f7dedcc - core::panicking::panic::hd695e3b1d0dd4ef4
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/panicking.rs:50:5
  16:     0x5625411b49cd - tikv_util::sys::cgroup::parse_mountinfos_v1::ha734f2e459f77064
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:184:17
  17:     0x5625411b49cd - tikv_util::sys::cgroup::cgroup_mountinfos_v1::hf403a30fb2a1bc54
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:174:5
  18:     0x5625411b49cd - tikv_util::sys::cgroup::CGroupSys::new::hccb0d17685fe4d7e
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/cgroup.rs:66:44
  19:     0x5625411b49cd - <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref::__static_ref_initialize::h28e3202687a0fbba
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/mod.rs:25:49
  20:     0x5625411b49cd - core::ops::function::FnOnce::call_once::h156e4e0161f03cad
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/ops/function.rs:227:5
  21:     0x5625411b49cd - lazy_static::lazy::Lazy<T>::get::{{closure}}::h4dde727c5a20cd1c
                               at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:31:29
  22:     0x5625411b49cd - std::sync::once::Once::call_once::{{closure}}::hb9879f936aec937a
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:261:41
  23:     0x56253f8374a8 - std::sync::once::Once::call_inner::hb74f8e4d768bb7d5
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:418:21
  24:     0x5625411b0e69 - std::sync::once::Once::call_once::h0c0db358d3797339
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sync/once.rs:261:9
  25:     0x5625411b0e69 - lazy_static::lazy::Lazy<T>::get::h586732d4b53595af
                               at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/inline_lazy.rs:30:9
  26:     0x5625411b0e69 - <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref::__stability::h038b756a132d4af3
                               at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:142:21
  27:     0x5625411b0e69 - <tikv_util::sys::SELF_CGROUP as core::ops::deref::Deref>::deref::h212d0313a92725af
                               at /rust/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:144:17
  28:     0x5625411b0e69 - tikv_util::sys::SysQuota::cpu_cores_quota::h84a8b5cdcb00fa43
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/components/tikv_util/src/sys/mod.rs:33:28
  29:     0x562540ddf81a - <tikv::config::UnifiedReadPoolConfig as core::default::Default>::default::h45e5a97432468693
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:1705:23
  30:     0x562540ddf81a - <tikv::config::ReadPoolConfig as core::default::Default>::default::h9b019deb51d1f626
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2008:5
  31:     0x562540ddf81a - <tikv::config::TiKvConfig as core::default::Default>::default::hdac27f1a6b612701
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2434:23
  32:     0x562540e587a0 - <tikv::config::_IMPL_DESERIALIZE_FOR_TiKvConfig::<impl serde::de::Deserialize for tikv::config::TiKvConfig>::deserialize::__Visitor as serde::de::Visitor>::visit_map::h2914076944cac18f
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2308:28
  33:     0x562540def2fb - <&mut toml::de::Deserializer as serde::de::Deserializer>::deserialize_any::h896098c68ffa07ea
                               at /rust/registry/src/github.com-1ecc6299db9ec823/toml-0.5.7/src/de.rs:220:19
  34:     0x562540def2fb - <&mut toml::de::Deserializer as serde::de::Deserializer>::deserialize_struct::hfd8bbc2f80c5453a
                               at /rust/registry/src/github.com-1ecc6299db9ec823/toml-0.5.7/src/de.rs:315:9
  35:     0x562540def2fb - tikv::config::_IMPL_DESERIALIZE_FOR_TiKvConfig::<impl serde::de::Deserialize for tikv::config::TiKvConfig>::deserialize::h38abcef80f3e5dd8
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2308:28
  36:     0x562540def2fb - tikv::config::TiKvConfig::from_file::{{closure}}::h7b83db76fd0099d1
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2829:17
  37:     0x562540def2fb - tikv::config::TiKvConfig::from_file::h5f0303e5cfb44775
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/src/config.rs:2823:9
  38:     0x56253f8ea2aa - tikv_server::main::{{closure}}::h66ecb6b13ede0646
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/cmd/tikv-server/src/main.rs:162:13
  39:     0x56253f8ea2aa - core::option::Option<T>::map_or_else::hfd350ffb5fd17fc4
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/option.rs:873:24
  40:     0x56253f8ea2aa - tikv_server::main::hcab930bf0217f089
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/cmd/tikv-server/src/main.rs:159:22
  41:     0x56253f8d3453 - core::ops::function::FnOnce::call_once::h9624d36abdf481d1
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/ops/function.rs:227:5
  42:     0x56253f8d3453 - std::sys_common::backtrace::__rust_begin_short_backtrace::h86a17e4f89429c5e
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:125:18
  43:     0x56253f8ec9c4 - main
  44:     0x7f9d77ffc3d5 - __libc_start_main
  45:     0x56253f8d2487 - <unknown>
  46:                0x0 - <unknown>
```

**Comment by sticnarf** [comment]
Sorry for the inconvenience. I think this is caused by a bug fixed on master: https://github.com/tikv/tikv/pull/11029

But the fix is not backported to the 5.2 release branch yet.

**Comment by jetora** [comment]
> Sorry for the inconvenience. I think this is caused by a bug fixed on master: #11029
> 
> But the fix is not backported to the 5.2 release branch yet.

Is there any plan to fix?

**Comment by sticnarf** [comment]
> > Sorry for the inconvenience. I think this is caused by a bug fixed on master: #11029
> > But the fix is not backported to the 5.2 release branch yet.
> 
> Is there any plan to fix?

Yes. It will be cherry picked to the 5.2 release in https://github.com/tikv/tikv/pull/11050 and will be included in 5.2.3. But I have no idea about when 5.2.3 will release (probably in half a month). So, if possible, you can try to compile yourself with that fix.

**Comment by sticnarf** [comment]
It's already merged into the 5.2 release branch. It will be brought out in the 5.2.4 release.
